### PR TITLE
Improve the Devfile is not reachable warning for SSH urls

### PR DESCRIPTION
**Pull request by vinokurig**
### What does this PR do?
If the devfile resolve request fails with the devfile not found error, check if the url is an SSH url. If so, interrupt the workspace start with a new warning that describes the problem of the devfile resolve via a private repository SSH url.

### What issues does this PR fix or reference?
https://issues.redhat.com/browse/CRW-7624

### Is it tested? How?

1. Add SSH keys to the user namespace: `User Preferences` -> `SSH Keys`
2. Start a workspace from an SSH url of a private repository.

See: the workspace start interrupts with the warning:
screenshot-cluster-admin-che-dashboard-local-server_apps_rosa_yz5n8-rzhzt-sum_vjy8_p3_openshiftapps_com-2024_11_19-13_20_31

A new documentation link is added to the warning.

### CODE REVIEW

**Comment by akurinnoy** [comment]
@vinokurig please take a look at the failed unit test

File: packages/dashboard-frontend/src/components/WorkspaceProgress/CreatingSteps/Fetch/Devfile/__tests__/index.spec.tsx [hunk 1]
```diff
@@ -742,6 +743,46 @@ describe('Creating steps, fetching a devfile', () => {
...
+
+    it('should show warning on SSH url', async () => {
+      const expectAlertItem = expect.objectContaining({
+        title: 'Warning',
+        variant: AlertVariant.warning,
+        children: (
+          <ExpandableWarning
+            textBefore="Devfile resolve from a privatre repositry via an SSH url is not supported."
+            errorMessage="Could not reach devfile"
+            textAfter="You can apply a Personal Access Token to be able to fetch the devfile.yaml content."
```
Comment (deerskindoll) [inline packages/dashboard-frontend/src/components/WorkspaceProgress/CreatingSteps/Fetch/Devfile/__tests__/index.spec.tsx:755]: if applying PAT solves the issue, I'd change the wording to this:

"Apply a Personal Access Token to fetch the devfile.yaml content."

Comment (vinokurig) [reply packages/dashboard-frontend/src/components/WorkspaceProgress/CreatingSteps/Fetch/Devfile/__tests__/index.spec.tsx]: Thanks, done

**Comment by dmytro-ndp** [comment]
@vinokurig : could you, please, take a look at the failing build https://github.com/eclipse-che/che-dashboard/actions/runs/11916044763/job/33207772337 ?

It prevents building dashboard image quay.io/eclipse/che-dashboard:pr-1259 needed to verify PR changes.

File: packages/dashboard-frontend/src/services/bootstrap/branding.constant.ts [hunk 2]
```diff
@@ -83,6 +84,8 @@ export const BRANDING_DEFAULT: BrandingData = {
       'https://www.eclipse.org/che/docs/stable/end-user-guide/url-parameter-for-the-workspace-storage/',
     webSocketTroubleshooting:
       'https://www.eclipse.org/che/docs/stable/end-user-guide/troubleshooting-network-problems/',
+    startingAWorkspaceFromAGitRepositoryURL:
```
Comment (dmytro-ndp) [inline packages/dashboard-frontend/src/services/bootstrap/branding.constant.ts:87]: Few remarks:
1) Variable name could be simplified and not linked to docs article name, which could be changed in future; e.g. `startWorkspaceFromGit`
2) Don't forget to add downstream documentation URL for `startingAWorkspaceFromAGitRepositoryURL` into the **product.json**: https://github.com/redhat-developer/devspaces-images/blob/devspaces-3-rhel-8/devspaces-dashboard/packages/dashboard-frontend/assets/branding/product.json#L21

**Review by olexii4 — APPROVED** [review]

File: packages/dashboard-frontend/src/services/bootstrap/branding.constant.ts (hunk 2 above)
Comment (vinokurig) [reply packages/dashboard-frontend/src/services/bootstrap/branding.constant.ts]: Done, pull request for the downstream: https://github.com/redhat-developer/devspaces-images/pull/658

**Comment by dmytro-ndp** [comment]
@vinokurig: the PR validated successfully using DS 3.17.0.RC-58 + dashboard deployed from quay.io/eclipse/che-dashboard:pr-1259.

Checked links by creating workspace using SSH URL of private repo having SSH key with and without passphrase added to the User Preferences:
1) "Continue with default devfile" -> workspace opened successfully with default devfile, and project cloned 
2) "Reload" - showed the same page
3) "Open Documentation page" -> lead to correct page
Screenshot from 2024-11-20 14-08-08

Public repo opened successfully from SSH URL. There were no any warnings at start workspace page.

Well done!

**Review by dmytro-ndp — APPROVED** [review]

**Comment by vinokurig** [comment]
If the devfile resolve request fails with the devfile not found error, check if the url is an SSH url. If so, interrupt the workspace start with a new warning that describes the problem of the devfile resolve via a private repository SSH url.

**Comment by devspacesbuild** [comment]
Build [3.18](https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-8/) :: [dashboard_3.x/592](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/dashboard_3.x/592/): [Console](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/dashboard_3.x/592/console), [Changes](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/dashboard_3.x/592/changes), [Git Data](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/dashboard_3.x/592/git)

**Comment by devspacesbuild** [comment]
Build [3.18](https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-8/) :: [sync-to-downstream_3.x/8119](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/sync-to-downstream_3.x/8119/): [Console](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/sync-to-downstream_3.x/8119/console), [Changes](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/sync-to-downstream_3.x/8119/changes), [Git Data](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/sync-to-downstream_3.x/8119/git)

**Comment by devspacesbuild** [comment]
Build [3.18](https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-8/) :: [get-sources-rhpkg-container-build_3.x/8196](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/get-sources-rhpkg-container-build_3.x/8196/): [Console](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/get-sources-rhpkg-container-build_3.x/8196/console), [Changes](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/get-sources-rhpkg-container-build_3.x/8196/changes), [Git Data](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/get-sources-rhpkg-container-build_3.x/8196/git)

**Pull request merged**